### PR TITLE
Major memory usage optimization in iEEG pipeline

### PR DESCRIPTION
**Pull request by menoua**

### CODE REVIEW

File: naplib/naplab/process_ieeg.py [hunk 1]
```diff
@@ -233,8 +238,8 @@ def process_ieeg(
     alignment_times = np.asarray(alignment_times) - earliest_time # shift times back since we are going to truncate the data
     earliest_sample = int(raw_data['data_f'] * earliest_time)
     latest_sample = int(raw_data['data_f'] * latest_time)
-    raw_data['data'] = raw_data['data'][earliest_sample:latest_sample]
-    raw_data['wav'] = raw_data['wav'][earliest_sample:latest_sample]
+    raw_data['data'] = raw_data['data'][earliest_sample:latest_sample].copy()
+    raw_data['wav'] = raw_data['wav'][earliest_sample:latest_sample].copy()
```
Comment (gavinmischler) [inline naplib/naplab/process_ieeg.py:242]: Why do you copy these here?

File: naplib/preprocessing/filter_hilbert.py [hunk 2]
```diff
@@ -161,23 +160,30 @@ def filterbank_hilbert(x, fs, Wn=[1,150], n_jobs=-1):
```
Comment (gavinmischler) [inline naplib/preprocessing/filter_hilbert.py:162]: Unless you add a parameter like ``return_average`` and set it to False by default, the above Examples section is now broken

File: naplib/naplab/process_ieeg.py (hunk 1 above)
Comment (menoua) [reply naplib/naplab/process_ieeg.py:242]: This was a part of the memory optimization. Returning a slice of the numpy array keeps the entire array alive in memory. Copying momentarily increases memory usage, but then drops the larger array and only keeps the smaller one. I couldn't find a way to discard the unwanted parts without making a copy.

File: naplib/preprocessing/filter_hilbert.py (hunk 2 above)
Comment (menoua) [reply naplib/preprocessing/filter_hilbert.py:162]: Separated this functionality into two functions now: `filter_hilbert` and `filterbank_hilbert`

**Review by gavinmischler — APPROVED** [review]

**Pull request merged**